### ISSUE TITLE
Correct type to use with id

### ISSUE THREAD

**Issue by awm33** [balderdashy/sails-mongo#232]
I switched an application from using MySQL to MongoDB.

If I try to find or update using a field `id`, it doesn't work. The field type was an integer, but looking at what gets sent to mongo, it shows up as NaN `{ _id: NaN }`. If I use a string type, it tries to use a regex `{ _id: /^54b955ffa52b17af67f2f347$/i }`. The object exists, but obviously NaN doesn't work and `{ _id: /^54b955ffa52b17af67f2f347$/i }` doesn't match anything but `{ _id: ObjectId('54b955ffa52b17af67f2f347') }` in the console does.

Is the above behavior expected?

How can I get it to convert `id` to `_id: ObjectId('54b955ffa52b17af67f2f347')` for the database query?

**Comment by Josebaseba** [comment]
Take a look that pull request: https://github.com/balderdashy/sails-mongo/pull/215 you can use that :+1:

**Comment by sailsbot** [comment]
Thanks for posting, @awm33.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
